### ISSUE TITLE
Restart Positions seem to not be respected on .MOD files

### ISSUE THREAD

**Issue by zeesworth** [libxmp/libxmp#763]
Hi, I'm trying out the example program with an example .MOD file I created (`player-openal-buffer` in specific, since that one doesn't stop after looping), and it seems like the module's Restart Position does not get respected when looping.

I've attached the file that shows what I mean, the file should loop back to the point where the screaming starts,
however when testing it with the example program it loops back to the beginning of the file.

[TEST.MOD.zip](https://github.com/user-attachments/files/17190610/TEST.MOD.zip)

Is there something I'm missing here? Or is this an issue with the library?

**Comment by AliceLR** [comment]
This appears to be an issue with libxmp. Your file is detected as an OpenMPT MOD early and skips the main tracker identification routine, which is the only place the restart position is set currently.

A word of caution with the MOD format: it is mess with regards to compatibility, and I recommend using pretty much any other format if possible. This field has multiple possible meanings depending on which tracker created it, and MOD offers nearly nothing in terms of determining which tracker made a module.

**Comment by zeesworth** [comment]
Oh dear, I didn't realize just how differently different tracker softwares made MOD files between each other.
I tested my example file with your changes and it works perfectly now, thank you

Unfortunately the use of MOD is a requirement for me, since the program I'm building uses the data from an external source, which uses MOD files to store some audio.
I had neglected to check any of the original MOD files, and in testing one of those files with the example program, it seems like the looping in this file always was correct.
I'm glad this managed to fix an issue, even if it wasn't the one I thought it was, thank you
